### DISCUSSION TITLE
How to install and configure fnm

### DISCUSSION THREAD

**Discussion by 0x7a7a** [zdharma-continuum/zinit#240] [Q&A] [answered]
I encountered two problems while installing [fnm](https://github.com/Schniz/fnm):
- I installed it successfully according to the example and my own understanding of the configuration, I would like to ask you to help me review(The main thing is the configuration of environment variables):
  ```shell
  zinit ice from'gh-r' sbin'fnm' nocompile\
      atclone'./fnm env --use-on-cd > fnmenv.zsh' \
      atpull"%atclone" \
      src"fnmenv.zsh"
  zinit light Schniz/fnm
  ```
- How to use output as zinit `snippet`? I want to configure fnm [completions](https://github.com/Schniz/fnm#completions) with the `fnm completions --shell zsh` command.  

I hope to get your help, thank you very much!

**Comment by 0x7a7a** [comment]
I changed to the `multisrc` method for source, and it works, but every time I open the terminal, an error appears
<img width="375" alt="image" src="https://user-images.githubusercontent.com/8832748/166108090-0543a386-3c1b-45c6-bde2-fb54c1dddb97.png">   

This is my modified configuration
```shell
zinit ice from'gh-r' sbin'fnm' nocompile \
    atclone"./fnm env --use-on-cd > fnmenv.zsh;./fnm completions --shell zsh > _fnm.zsh" \
    atpull"%atclone" \
    multisrc"*.zsh"
zinit light Schniz/fnm
```

**Comment by 0x7a7a** [comment]
I found some [information](https://github.com/Schniz/fnm/issues/491), it seems to be because of `fpath`, but I don't know how to deal with the best

**Comment by vladdoster** [comment]
```zsh
zinit for \
    as'completion' \
    atclone"
      ./fnm env --use-on-cd > fnmenv.zsh
      ./fnm completions --shell zsh > _fnm.zsh" \
    atpull'%atclone' \
    blockf \
    from'gh-r' \
    nocompile \
    pick'fnmenv.zsh' \
    sbin'fnm' \
  @Schniz/fnm
```

<img width="1475" alt="Screen Shot 2022-04-30 at 15 26 54" src="https://user-images.githubusercontent.com/10052309/166121548-628aa29c-a657-498e-9953-7f5266f0c557.png">

**Reply by 0x7a7a** [reply]
Thank you so much for your help!
It seems I missed some details, I learned a lot from your configuration

**Comment by vladdoster** [comment]
Btw, I've added fnm to the [wiki](https://github.com/zdharma-continuum/zinit/wiki/%F0%9F%A7%8A-Recommended-ices#fnm) and will add a zunit test.

**Reply by 0x7a7a** [reply]
Great, I think this can help newbies like me:)

**Comment by 0x7a7a** [comment]
@vladdoster 
I'm sorry, but I found a new problem while using the script you suggested.
It looks like the environment variables are not added correctly, so it doesn't work.
<img width="1003" alt="image" src="https://user-images.githubusercontent.com/8832748/166123273-2374fe69-fa80-4383-aa98-6f2e8164647c.png">

**Comment by vladdoster** [comment, marked as answer]
```zsh
zinit for \
    as'completion' \
    atclone"./fnm completions --shell zsh > _fnm.zsh" \
    atload'eval $(fnm env --shell zsh)' \
    atpull'%atclone' \
    blockf \
    from'gh-r' \
    nocompile \
    sbin'fnm' \
  @Schniz/fnm
```

<img width="2144" alt="Screen Shot 2022-04-30 at 17 58 24" src="https://user-images.githubusercontent.com/10052309/166125332-5ee818bc-d5a7-4508-b7b4-e768bad83381.png">

**Reply by 0x7a7a** [reply]
Thank you very much, it looks perfect now

**Reply by alexgorbatchev** [reply]
This doesn't appear to work any more, getting the following error now:

```
Downloading Schniz/fnm…
Didn't find correct Github release-file to download.
```

Does anyone know how to fix this?

**Reply by vladdoster** [reply]
@alexgorbatchev,

It still works. I can guarantee it works for macOS and Linux x86_64 because there is a[ unit test for fnm](https://github.com/zdharma-continuum/zinit/blob/cbb4a880ae1fae86def016e68d783c02f1a00dab/tests/gh-r.zunit#L202-L206).

<img width="716" alt="Screenshot 2023-04-10 at 17 20 28" src="https://user-images.githubusercontent.com/10052309/231010123-843fb3c0-4c56-4c4a-8708-f314530baafa.png">

To help resolve your problem, please open a new issue instead of commenting on the answer of a year old discussion.

<img width="884" alt="Screenshot 2023-04-10 at 16 57 11" src="https://user-images.githubusercontent.com/10052309/231007048-8b8c4be9-9772-4dc1-b83f-ca0e2d894ee9.png">
